R/hn_item.R

```diff
@@ -0,0 +1,55 @@
+#' @title Print for "hn_item" type objects
+#'
+#' @param x "hn_item" type list
+#'
+#' @param ... further arguments passed to or from other methods
+#'
+#' @export
+#'
+#' @examples
+#'
+#' hacker_news <- get_top_stories(max_items = 5)
+#' print(hacker_news)
+
+print.hn_item <- function(x, ...) {
+  utils::str(x)
+}
+
+#' Creates an object representing Hacker News item
+#'
+#' @param item list returned by Hacker News API
+#'
+#' @return object represanting a hn_item
+#'
+create_hn_item <- function(item) {
+  if (is.null(item)) {
+    return(NULL)
+  }
+
+  if (is.numeric(item$time)) {
+    item$time <- as.POSIXct(item$time, origin = "1970-01-01")
```

Comment:
What is this line for?

Reply:
- to avoid converting not numeric object
- easier to use time objects for data aggregation etc.

Reply:
- Do you mean that sometimes the time get's parsed as a numeric and sometimes as something else by `jsonlite`?
- I agree, but what it the `origin` argument for?

Reply:
- I don't know whether it will always return numeric, but if not, we can just leave time as it is, because only id is required.
- origin is required if we want to convert numeric -> time
as.POSIXct(item$time) will result with error

Reply:
- I would ommit the if and always convert it. It seems like the API returns the time always in the same way so it will be parsed the same way as well.
- You are right. I didn't know about that, maybe let's wrap the value in a function like `datetime_origin` and supply a link to a source that explains why it is needed :)